src/v0/destinations/sfmc/transform.test.js

```diff
@@ -0,0 +1,179 @@
+        subDomain: 'yourSubDomain',
+        createOrUpdateContacts: false,
+        externalKey: 'yourExternalKey',
+        eventToExternalKey: [{ from: 'purchase', to: 'purchaseKey' }],
+        eventToPrimaryKey: [{ from: 'purchase', to: 'primaryKey' }],
+        eventToUUID: [{ event: 'purchase', uuid: true }],
+      },
+    };
+
+    try {
+      await responseBuilderSimple(message, category, destination);
+    } catch (e) {
+      expect(e).toBeInstanceOf(ConfigurationError);
+      expect(e.message).toBe('Event name is required for this track events');
+    }
+  });
+
+  // Throws an error when event is not mapped for track calls
+  it('should throw an error when event is not mapped for track calls', async () => {
+    const message = {
+      type: 'track',
+      event: 'unmappedEvent',
+    };
+
+    const category = {
+      type: 'track',
+      name: 'Track',
+    };
+
+    const destination = {
+      Config: {
+        clientId: 'yourClientId',
+        clientSecret: 'yourClientSecret',
+        subDomain: 'yourSubDomain',
+        createOrUpdateContacts: false,
+        externalKey: 'yourExternalKey',
+        eventToExternalKey: [{ from: 'purchase', to: 'purchaseKey' }],
+        eventToPrimaryKey: [{ from: 'purchase', to: 'primaryKey' }],
+        eventToUUID: [{ event: 'purchase', uuid: true }],
+      },
+    };
+
+    try {
+      await responseBuilderSimple(message, category, destination);
+    } catch (e) {
+      expect(e).toBeInstanceOf(ConfigurationError);
+      expect(e.message).toBe('Event not mapped for this track call');
+    }
+  });
+
+  // Throws an error when event type is not supported
+  it('should throw an error when event type is not supported', async () => {
+    const message = {
+      type: 'unsupported',
+    };
+
+    const category = {
+      type: 'unsupported',
+      name: 'Unsupported',
+    };
+
+    const destination = {
+      Config: {
+        clientId: 'yourClientId',
+        clientSecret: 'yourClientSecret',
+        subDomain: 'yourSubDomain',
+        createOrUpdateContacts: false,
+        externalKey: 'yourExternalKey',
+        eventToExternalKey: [{ from: 'purchase', to: 'purchaseKey' }],
+        eventToPrimaryKey: [{ from: 'purchase', to: 'primaryKey' }],
+        eventToUUID: [{ event: 'purchase', uuid: true }],
+      },
+    };
+
+    try {
+      await responseBuilderSimple(message, category, destination);
+    } catch (e) {
+      expect(e).toBeInstanceOf(ConfigurationError);
+      expect(e.message).toBe('Event type not supported');
+    }
+  });
+
+  // Returns a payload for track calls when event is mapped and event name is a string
+  it('should return a payload for track calls when event is mapped and event name is a string', async () => {
+    const message = {
+      type: 'track',
+      event: 'purchase',
+      userId: '12345',
+    };
+
+    const category = {
+      type: 'track',
+      name: 'Track',
+    };
+
+    const destination = {
```

Comment:
can make this global const if no changes required across test cases

Reply:
Addressed